File: converted_source/uborders.pas.py
# unit uborders

from conversion_common import *
import usupport
import updform
import udomain
import delphi_compatability

# const
kTransferLoad = 1
kTransferSave = 2

class TPrintBordersForm(PdForm):
    def __init__(self):
        self.Close = TButton()
        self.cancel = TButton()
        self.printBorderInner = TCheckBox()
        self.printBorderOuter = TCheckBox()
        self.printBorderColorInnerLabel = TLabel()
        self.printBorderColorInner = TPanel()
        self.printBorderColorOuter = TPanel()
        self.printBorderColorOuterLabel = TLabel()
        self.printBorderWidthInnerLabel = TLabel()
        self.printBorderWidthOuterLabel = TLabel()
        self.printBorderWidthInner = TSpinEdit()
        self.printBorderWidthOuter = TSpinEdit()
        self.helpButton = TSpeedButton()
        self.options = BitmapOptionsStructure()
        self.transferDirection = 0
        self.optionsChanged = false
    
    #$R *.DFM
    def initializeWithOptions(self, anOptions):
        self.options = anOptions
        self.transferDirection = kTransferLoad
        self.transferFields()
        # simple interaction, so just put it in here
        self.printBorderColorInner.Enabled = self.printBorderInner.Checked
        self.printBorderWidthInner.Enabled = self.printBorderInner.Checked
        self.printBorderColorInnerLabel.Enabled = self.printBorderInner.Checked
        self.printBorderWidthInnerLabel.Enabled = self.printBorderInner.Checked
        self.printBorderColorOuter.Enabled = self.printBorderOuter.Checked
        self.printBorderWidthOuter.Enabled = self.printBorderOuter.Checked
        self.printBorderColorOuterLabel.Enabled = self.printBorderOuter.Checked
        self.printBorderWidthOuterLabel.Enabled = self.printBorderOuter.Checked
    
    def transferFields(self):
        self.options.printBorderInner = self.transferCheckBox(self.printBorderInner, self.options.printBorderInner)
        self.options.printBorderOuter = self.transferCheckBox(self.printBorderOuter, self.options.printBorderOuter)
        self.transferColor(self.printBorderColorInner, self.options.printBorderColorInner)
        self.transferColor(self.printBorderColorOuter, self.options.printBorderColorOuter)
        self.options.printBorderWidthInner = self.transferSmallintSpinEditBox(self.printBorderWidthInner, self.options.printBorderWidthInner)
        self.options.printBorderWidthOuter = self.transferSmallintSpinEditBox(self.printBorderWidthOuter, self.options.printBorderWidthOuter)
    
    def transferCheckBox(self, checkBox, value):
        if self.transferDirection == kTransferLoad:
            checkBox.Checked = value
        elif self.transferDirection == kTransferSave:
            if value != checkBox.Checked:
                self.optionsChanged = true
            value = checkBox.Checked
        return value
    
    def transferColor(self, colorPanel, value):
        raise "method transferColor had assigned to var parameter value not added to return; fixup manually"
        if self.transferDirection == kTransferLoad:
            colorPanel.Color = value
        elif self.transferDirection == kTransferSave:
            if value != colorPanel.Color:
                self.optionsChanged = true
            value = colorPanel.Color
    
    def transferSmallintSpinEditBox(self, editBox, value):
        if self.transferDirection == kTransferLoad:
            editBox.Text = IntToStr(value)
        elif self.transferDirection == kTransferSave:
            if value != StrToIntDef(editBox.Text, 0):
                self.optionsChanged = true
            value = StrToIntDef(editBox.Text, 0)
        return value
    
    def printBorderColorInnerMouseUp(self, Sender, Button, Shift, X, Y):
        self.printBorderColorInner.Color = udomain.domain.getColorUsingCustomColors(self.printBorderColorInner.Color)
    
    def printBorderColorOuterMouseUp(self, Sender, Button, Shift, X, Y):
        self.printBorderColorOuter.Color = udomain.domain.getColorUsingCustomColors(self.printBorderColorOuter.Color)
    
    def printBorderInnerClick(self, Sender):
        self.printBorderColorInner.Enabled = self.printBorderInner.Checked
        self.printBorderWidthInner.Enabled = self.printBorderInner.Checked
        self.printBorderColorInnerLabel.Enabled = self.printBorderInner.Checked
        self.printBorderWidthInnerLabel.Enabled = self.printBorderInner.Checked
    
    def printBorderOuterClick(self, Sender):
        self.printBorderColorOuter.Enabled = self.printBorderOuter.Checked
        self.printBorderWidthOuter.Enabled = self.printBorderOuter.Checked
        self.printBorderColorOuterLabel.Enabled = self.printBorderOuter.Checked
        self.printBorderWidthOuterLabel.Enabled = self.printBorderOuter.Checked
    
    def CloseClick(self, Sender):
        self.transferDirection = kTransferSave
        self.transferFields()
        self.ModalResult = mrOK
    
    def cancelClick(self, Sender):
        self.ModalResult = mrCancel
    
    def helpButtonClick(self, Sender):
        delphi_compatability.Application.HelpJump("Drawing_a_print_border")
    
    def FormKeyPress(self, Sender, Key):
        Key = usupport.makeEnterWorkAsTab(self, self.ActiveControl, Key)
        return Key
    

File: converted_source/uborders.pas.java
// unit uborders

from conversion_common import *;
import updform;
import delphi_compatability;

// const
kTransferLoad = 1;
kTransferSave = 2;



class TPrintBordersForm extends PdForm {
    public TButton Close;
    public TButton cancel;
    public TCheckBox printBorderInner;
    public TCheckBox printBorderOuter;
    public TLabel printBorderColorInnerLabel;
    public TPanel printBorderColorInner;
    public TPanel printBorderColorOuter;
    public TLabel printBorderColorOuterLabel;
    public TLabel printBorderWidthInnerLabel;
    public TLabel printBorderWidthOuterLabel;
    public TSpinEdit printBorderWidthInner;
    public TSpinEdit printBorderWidthOuter;
    public TSpeedButton helpButton;
    public BitmapOptionsStructure options;
    public short transferDirection;
    public boolean optionsChanged;
    
    //$R *.DFM
    public void initializeWithOptions(BitmapOptionsStructure anOptions) {
        this.options = anOptions;
        this.transferDirection = kTransferLoad;
        this.transferFields();
        // simple interaction, so just put it in here
        this.printBorderColorInner.Enabled = this.printBorderInner.Checked;
        this.printBorderWidthInner.Enabled = this.printBorderInner.Checked;
        this.printBorderColorInnerLabel.Enabled = this.printBorderInner.Checked;
        this.printBorderWidthInnerLabel.Enabled = this.printBorderInner.Checked;
        this.printBorderColorOuter.Enabled = this.printBorderOuter.Checked;
        this.printBorderWidthOuter.Enabled = this.printBorderOuter.Checked;
        this.printBorderColorOuterLabel.Enabled = this.printBorderOuter.Checked;
        this.printBorderWidthOuterLabel.Enabled = this.printBorderOuter.Checked;
    }
    
    public void transferFields() {
        this.options.printBorderInner = this.transferCheckBox(this.printBorderInner, this.options.printBorderInner);
        this.options.printBorderOuter = this.transferCheckBox(this.printBorderOuter, this.options.printBorderOuter);
        this.transferColor(this.printBorderColorInner, this.options.printBorderColorInner);
        this.transferColor(this.printBorderColorOuter, this.options.printBorderColorOuter);
        this.options.printBorderWidthInner = this.transferSmallintSpinEditBox(this.printBorderWidthInner, this.options.printBorderWidthInner);
        this.options.printBorderWidthOuter = this.transferSmallintSpinEditBox(this.printBorderWidthOuter, this.options.printBorderWidthOuter);
    }
    
    public void transferCheckBox(TCheckBox checkBox, boolean value) {
        if (this.transferDirection == kTransferLoad) {
            checkBox.Checked = value;
        } else if (this.transferDirection == kTransferSave) {
            if (value != checkBox.Checked) {
                this.optionsChanged = true;
            }
            value = checkBox.Checked;
        }
        return value;
    }
    
    public void transferColor(TPanel colorPanel, TColorRef value) {
        raise "method transferColor had assigned to var parameter value not added to return; fixup manually"
        if (this.transferDirection == kTransferLoad) {
            colorPanel.Color = value;
        } else if (this.transferDirection == kTransferSave) {
            if (value != colorPanel.Color) {
                this.optionsChanged = true;
            }
            value = colorPanel.Color;
        }
    }
    
    public void transferSmallintSpinEditBox(TSpinEdit editBox, short value) {
        if (this.transferDirection == kTransferLoad) {
            editBox.Text = IntToStr(value);
        } else if (this.transferDirection == kTransferSave) {
            if (value != StrToIntDef(editBox.Text, 0)) {
                this.optionsChanged = true;
            }
            value = StrToIntDef(editBox.Text, 0);
        }
        return value;
    }
    
    public void printBorderColorInnerMouseUp(TObject Sender, TMouseButton Button, TShiftState Shift, int X, int Y) {
        this.printBorderColorInner.Color = UNRESOLVED.domain.getColorUsingCustomColors(this.printBorderColorInner.Color);
    }
    
    public void printBorderColorOuterMouseUp(TObject Sender, TMouseButton Button, TShiftState Shift, int X, int Y) {
        this.printBorderColorOuter.Color = UNRESOLVED.domain.getColorUsingCustomColors(this.printBorderColorOuter.Color);
    }
    
    public void printBorderInnerClick(TObject Sender) {
        this.printBorderColorInner.Enabled = this.printBorderInner.Checked;
        this.printBorderWidthInner.Enabled = this.printBorderInner.Checked;
        this.printBorderColorInnerLabel.Enabled = this.printBorderInner.Checked;
        this.printBorderWidthInnerLabel.Enabled = this.printBorderInner.Checked;
    }
    
    public void printBorderOuterClick(TObject Sender) {
        this.printBorderColorOuter.Enabled = this.printBorderOuter.Checked;
        this.printBorderWidthOuter.Enabled = this.printBorderOuter.Checked;
        this.printBorderColorOuterLabel.Enabled = this.printBorderOuter.Checked;
        this.printBorderWidthOuterLabel.Enabled = this.printBorderOuter.Checked;
    }
    
    public void CloseClick(TObject Sender) {
        this.transferDirection = kTransferSave;
        this.transferFields();
        this.ModalResult = mrOK;
    }
    
    public void cancelClick(TObject Sender) {
        this.ModalResult = mrCancel;
    }
    
    public void helpButtonClick(TObject Sender) {
        delphi_compatability.Application.HelpJump("Drawing_a_print_border");
    }
    
    public void FormKeyPress(TObject Sender, char Key) {
        UNRESOLVED.makeEnterWorkAsTab(this, this.ActiveControl, Key);
        return Key;
    }
    
}
